### ISSUE TITLE
Allow installing themes and plugins via file upload

### ISSUE THREAD

**Issue by flaviocopes** [getgrav/grav-plugin-admin#895]

**Comment by OleVik** [comment]
Would this not require a layer of checking for malware upon upload? Intermediate developers might be confident in getting safe themes/plugins, but a novice might upload something that breaks the site and/or server. If there already is a mechanism for this, then pardon my ignorance.

**Comment by flaviocopes** [comment]
I don't think this belongs to the responsibilities of the uploader, but to the user to obtain packages from the right sources. We cannot possibly deeply cover malware scanning, not much more than scanning for eval / base64 usage for example.

**Comment by OleVik** [comment]
Common things like eval, base64, or other shell- or code-injections would be sufficient as they are the only real targets for Grav itself. Broader server-attacks like malware should be caught by the users own antivirus or the hosts detection-measures.

A [quick list of PHP-functions](https://www.owasp.org/index.php/PHP_Security_Cheat_Sheet#Other_Injection_Cheat_Sheet) to check for.
